ecosystem/typescript/sdk_v2/src/crypto/authentication_key.ts

```diff
@@ -0,0 +1,83 @@
+// Copyright © Aptos Foundation
+// SPDX-License-Identifier: Apache-2.0
+
+import { sha3_256 as sha3Hash } from "@noble/hashes/sha3";
+import { AccountAddress, Hex } from "../core";
+import { HexInput } from "../types";
+import { MultiEd25519PublicKey } from "./multi_ed25519";
+import { Ed25519PublicKey } from "./ed25519";
+
+/**
+ * Each account stores an authentication key. Authentication key enables account owners to rotate
+ * their private key(s) associated with the account without changing the address that hosts their account.
+ * @see {@link * https://aptos.dev/concepts/accounts | Account Basics}
+ *
+ * Account addresses can be derived from AuthenticationKey
+ */
+export class AuthenticationKey {
+  // Length of AuthenticationKey in bytes(UInt8Array)
+  static readonly LENGTH: number = 32;
+
+  // Scheme identifier for MultiEd25519 signatures used to derive authentication keys for MultiEd25519 public keys
+  static readonly MULTI_ED25519_SCHEME: number = 1;
+
+  // Scheme identifier for Ed25519 signatures used to derive authentication key for MultiEd25519 public key
+  static readonly ED25519_SCHEME: number = 0;
+
+  // Scheme identifier used when hashing an account's address together with a seed to derive the address (not the
+  // authentication key) of a resource account.
+  static readonly DERIVE_RESOURCE_ACCOUNT_SCHEME: number = 255;
+
+  private readonly _data: Hex;
```

Comment:
since we represent the data as `Hex`, should we also support `toUint8Array()` and `toString()` methods for easy retrieval? 
```
 toString(): string {
    return `0x${this.data.toString()}`;
  }

toUint8Array(): Uint8Array {
    return this.data.toUint8Array();
  }
```

Reply:
Good call!